release.sh

```diff
@@ -1,23 +1,27 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-set -v -e -o pipefail
+set -eux -o pipefail
 
 VERSION=$(git describe --abbrev=0 --tags)
 ARCH="${ARCH:-64}"
-CUSTOM_FLAGS=""
-
+CUSTOM_FLAGS=()
 unameOut="$(uname -s)"
 case "$unameOut" in
     Linux*)
         OS=linux
-        CUSTOM_FLAGS="-L--export-dynamic"
+        CUSTOM_FLAGS+=("-L--export-dynamic")
         ;;
     Darwin*)
         OS=osx
+        CUSTOM_FLAGS+=("-L-macosx_version_min" "-L10.7" "-L-lcrt1.o")
         ;;
     *) echo "Unknown OS: $unameOut"; exit 1
 esac
 
+if [[ $(basename "$DMD") =~ ldmd.* ]] ; then
+    CUSTOM_FLAGS+=("-flto=full" "-linker=gold")
```

Comment:
Ported the configuration from https://github.com/dlang-community/D-Scanner/pull/598 to this script.
(the binaries are here https://github.com/dlang-community/D-Scanner/releases/tag/v0.5.1)
Note: LDC's `-static` linking can't be used because `curl` (it's `dlopen-ed` in std.net.curl and apparently is `std.socket` not very static link-friendly either)

Reply:
Perhaps we should have a version statement that allows to statically link curl. Then we could use the musl C standard library that fully supports static linking.